components/Table/Table.js

```diff
@@ -95,21 +162,148 @@ export class Table {
     this.tableHeader.append(tableHeaderRow);
   }
 
+  filterMenu() {
+    const filterMenuContainer = document.createElement('div');
+    filterMenuContainer.classList.add('filter-menu-container');
+
+    const filterLabel = document.createElement('div');
+    filterLabel.classList.add('menu-label');
+    filterLabel.textContent = 'Name';
+
+    const filterOperator = document.createElement('div');
+    filterOperator.classList.add('filter-operator');
+    filterOperator.textContent = 'equal';
+
+    const filterValue = document.createElement('input');
+    filterValue.classList.add('filter-value');
+    filterValue.placeholder = 'Filter value';
+
+    const btnClean = document.createElement('div');
+    btnClean.classList.add('btn-clean');
+    btnClean.textContent = 'x';
+
+    const btnExit = document.createElement('div');
+    btnExit.classList.add('btn-exit');
+    btnExit.textContent = 'x';
+
+    btnExit.addEventListener('click', () => {
+      filterMenuContainer.style.display = 'none';
+    });
+
+    btnClean.addEventListener('click', () => {
+      filterValue.value = '';
+      document
+        .querySelectorAll('.table-row')
+        .forEach((row) => (row.style.display = ''));
+    });
+
+    const filterFunction = (e) => {
+      this.paginationCallback();
+      let filter = e.target.value;
+      for (let key in this.tableElementsMap) {
+        if (key === 'name') {
+          this.tableElementsMap[key].forEach((item) =>
+            item.innerHTML.toLowerCase().indexOf(filter) > -1
+              ? (item.parentElement.style.display = '')
+              : (item.parentElement.style.display = 'none')
+          );
+        }
+      }
+    };
+
+    filterValue.addEventListener('input', filterFunction);
```

Comment:
We should not filter current values
This logic should work the same as we have for our sorting
We'll filter outer data and then render new data in the table